azurerm/resource_arm_kusto_eventhub_data_connection.go

```diff
@@ -0,0 +1,300 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/kusto/mgmt/2019-05-15/kusto"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmKustoEventHubDataConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKustoEventHubDataConnectionCreateUpdate,
+		Read:   resourceArmKustoEventHubDataConnectionRead,
+		Update: resourceArmKustoEventHubDataConnectionCreateUpdate,
+		Delete: resourceArmKustoEventHubDataConnectionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoDataConnectionName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"cluster_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoClusterName,
+			},
+
+			"database_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMKustoDatabaseName,
+			},
+
+			"eventhub_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"consumer_group": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateEventHubConsumerName(),
+			},
+
+			"table_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAzureRMKustoEntityName,
+			},
+
+			"mapping_rule_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAzureRMKustoEntityName,
+			},
+
+			"data_format": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAzureRMKustoDataConnectionDataFormat,
```

Comment:
rather than using the `All` method could we use:

```
ValidateFunc: validation.StringInSlice([]string{
  string(kusto.Thing),
}, false),
```

whilst the "All" methods are convenient it's possible values will be missed in the documentation if they're added to the SDK - so until we can generate those could we hard-code this for the moment?

Reply:
done